utils/_context/core.py

```diff
@@ -70,8 +78,12 @@ def serialize(self):
             "appsec_rules_file": self.appsec_rules_file or "*default*",
             "uds_socket": self.uds_socket,
             "scenario": self.scenario,
+            "components": self.components,
         }
 
+        if self.parametrized_tests_metadata:
```

Comment:
suggestion : 

```suggestion
    results |= self.components
```

And make any needed change on this object later.

Reply:
applied for components, no for parametrized_tests_metadata